From: Al Viro <viro@zeniv.linux.org.uk>
Date: Mon, 7 Mar 2016 15:11:43 -0500
Subject: ceph_fill_trace(): don't bother with d_instantiate(dn, NULL)
Git-commit: f8b31710e46c852b2af1635d1b6fab8e69bf7799
Patch-mainline: v4.6-rc1
References: FATE#322288

... and use d_add(dn, NULL) in case we need to hash a negative
unhashed rather than using d_rehash() directly.

Signed-off-by: Al Viro <viro@zeniv.linux.org.uk>
Acked-by: Luis Henriques <lhenriques@suse.com>
---
 fs/ceph/inode.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/fs/ceph/inode.c b/fs/ceph/inode.c
index 05dd66fe23a2..be2d87f33177 100644
--- a/fs/ceph/inode.c
+++ b/fs/ceph/inode.c
@@ -1236,10 +1236,8 @@ int ceph_fill_trace(struct super_block *sb, struct ceph_mds_request *req,
 				dout("d_delete %p\n", dn);
 				d_delete(dn);
 			} else {
-				dout("d_instantiate %p NULL\n", dn);
-				d_instantiate(dn, NULL);
 				if (have_lease && d_unhashed(dn))
-					d_rehash(dn);
+					d_add(dn, NULL);
 				update_dentry_lease(dn, rinfo->dlease,
 						    session,
 						    req->r_request_started);

